bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -306,25 +357,77 @@ impl Client {
-                homeserver_url,
-                sliding_sync_proxy,
-            })
+            match auth_api {
+                // Build the session from the regular Matrix Auth Session.
+                AuthApi::Matrix(a) => {
+                    let matrix_sdk::matrix_auth::Session {
+                        meta: matrix_sdk::SessionMeta { user_id, device_id },
+                        tokens:
+                            matrix_sdk::matrix_auth::SessionTokens { access_token, refresh_token },
+                    } = a.session().context("Missing session")?;
+
+                    Ok(Session {
+                        access_token,
+                        refresh_token,
+                        user_id: user_id.to_string(),
+                        device_id: device_id.to_string(),
+                        homeserver_url,
+                        oidc_data: None,
+                        sliding_sync_proxy,
+                    })
+                }
+                // Build the session from the OIDC UserSession.
+                AuthApi::Oidc(api) => {
+                    let matrix_sdk::oidc::UserSession {
+                        meta: matrix_sdk::SessionMeta { user_id, device_id },
+                        tokens:
+                            matrix_sdk::oidc::SessionTokens {
+                                access_token,
+                                refresh_token,
+                                latest_id_token,
+                            },
+                        issuer_info,
+                    } = api.user_session().context("Missing session")?;
+                    let client_id = api
+                        .client_credentials()
+                        .context("OIDC client credentials are missing.")?
+                        .client_id()
+                        .to_owned();
+                    let client_metadata =
+                        api.client_metadata().context("OIDC client metadata is missing.")?.clone();
+                    let oidc_data = OidcSessionData {
+                        client_id,
+                        client_metadata,
+                        latest_id_token: latest_id_token.map(|t| t.to_string()),
+                        issuer_info,
+                    };
+
+                    let oidc_data = serde_json::to_string(&oidc_data).ok();
+                    Ok(Session {
+                        access_token,
+                        refresh_token,
+                        user_id: user_id.to_string(),
+                        device_id: device_id.to_string(),
+                        homeserver_url,
+                        oidc_data,
+                        sliding_sync_proxy,
+                    })
+                }
+                _ => Err(anyhow!("Unknown authentication API").into()),
+            }
         })
     }
 
+    pub fn account_url(&self) -> Option<String> {
+        self.inner.oidc().account_management_url().unwrap_or(None).map(|url| url.to_string())
```

Comment:
A bit simpler:

```suggestion
        self.inner.oidc().account_management_url().ok().map(ToString::to_string)
```

Reply:
This one doesn't work for me, it returns the following error:

```
error[E0631]: type mismatch in function arguments
```

Closes I can get to work is

```
self.inner.oidc().account_management_url().ok().flatten().map(|url| url.to_string())
```

But that doesn't seem particularly better than the current version to me.